tools/performance-server/src/main/kotlin/database/BenchmarksIndexesDispatcher.kt

```diff
@@ -0,0 +1,323 @@
+                                ${buildNumbers.map { "\"$it\": { \"match\" : { \"buildNumber\" : \"$it\" }}" }
+                    .joinToString(",\n")}
+                            }
+                        },"""
+        } ?: ""}
+                    "aggs" : {
+                        "metric_build" : {
+                            "nested" : {
+                                "path" : "benchmarks"
+                            },
+                            "aggs" : {
+                                "metric_samples": {
+                                    "filters" : { 
+                                        "filters": { "samples": { $filterBenchmarks } }
+                                    },
+                                    "aggs" : {
+                                        "sum_log_x": {
+                                            "sum": {
+                                                "field" : "benchmarks.${if (normalize) "normalizedScore" else "score"}",
+                                                "script" : {
+                                                    "source": "if (_value == 0) { 0.0 } else { Math.log(_value) }"
+                                                }
+                                            }
+                                        },
+                                        "geom_mean": {
+                                            "bucket_script": {
+                                                "buckets_path": {
+                                                    "sum_log_x": "sum_log_x",
+                                                    "x_cnt": "_count"
+                                                },
+                                                "script": "Math.exp(params.sum_log_x/params.x_cnt)"
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        
+                           ${buildNumbers?.let {
+            """ }
+                        }"""
+        } ?: ""}
+                    }
+                }
+            }
+        """
+
+        return getIndex(featureValue).search(queryDescription, listOf("aggregations")).then { responseString ->
+            val dbResponse = JsonTreeParser.parse(responseString).jsonObject
+            val aggregations = dbResponse.getObjectOrNull("aggregations") ?: error("Wrong response:\n$responseString")
+            buildNumbers?.let {
+                val buckets = aggregations.getObjectOrNull("builds")?.getObjectOrNull("buckets")
+                        ?: error("Wrong response:\n$responseString")
+                buildNumbers.map {
+                    it to listOf(buckets.getObject(it).getObject("metric_build").getObject("metric_samples").getObject("buckets").getObject("samples").getObject("geom_mean").getPrimitive("value").double)
```

Comment:
I think it's better to make such chains multiline. See https://kotlinlang.org/docs/reference/coding-conventions.html#chained-call-wrapping